### DISCUSSION TITLE
How to extend the current inline ":class" conditions within encapsulation

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3521] [1. Help]
I have a very [simple encapsulation](https://jsfiddle.net/75cu29je/), which adds an animation class to the html element when it gets visible.

However, the problem is, when the element already has a `:class` condition, it conflicts/overrides. Sure, I could pass the existing condition as a parameter to the encapsulation, but this would make to code less readable in my opinion. Is there an elegant way to just combine these conditions?

PS: I know, occupying the `x-bind` for this simple encapsulation is probably a bit an overhead, probably it would be better to create a custom directive, but I'm not yet that deep into AlpineJS to go there I guess.

**Comment by ekwoka** [comment]
bind is actually a fine way to do it, probably doesn't need an `Alpine.data`. Just use `Alpine.bind`.

This shouldn't conflict...I'll look into it

Okay, it is a conflict, because of how the binding of a class works. x-bind of the object doesn't cause the issue, it creates separate bindings, allowing both to coexist. However, binding of a class also involves a step of removing any classes added in the last binding. Both binding would still be attached, but they'd constantly be fighting with each other.

Interesting. I'm not sure if the solution to this would be simple. I think at least a bit of the bind logic would need to be refactored to give them separate control over their own classes...and maybe even more to avoid issues where two class bindings are wanting to control the same class

**Reply by simonferndriger** [reply]
Thanks for the hint about `Alpine.bind`, I'm sure I can use that for some other cases. I'm interested to see however, where this could be going.

Happy Easter!

**Reply by simonferndriger** [reply]
I actually prefer `Alpine.data` over `Alpine.bind` because there I have a local scope :)

**Comment by simonferndriger** [comment]
"Solved" this issue by using a dedicated html attribute `data-animate="[effect]"` instead of a new class
